components/TopBarProfileMenu.js

```diff
@@ -155,16 +155,18 @@ class TopBarProfileMenu extends React.Component {
   }
 
   componentDidMount() {
-    document.addEventListener('keydown', this.handleKeyPress);
-    document.addEventListener('click', this.onClickOutside);
+    const main = document.querySelector('main');
+    main.addEventListener('keydown', this.handleKeyPress);
+    main.addEventListener('click', this.onClickOutside);
```

Comment:
Why?

Reply:
The reason to do this here is that note that targeting `document` means the menu will close whenever you click anywhere on the document; including within the menu itself. We don't want that; especially now that we have a drop-down within the menu when we click on the dropdown the menu shouldn't close. Ideally in general I don't think a menu should close when you click within the menu; only when the user clicks outside of the menu. To further illustrate this see the following screen-casts.

Before:

[before-dropdown.webm](https://user-images.githubusercontent.com/12435965/209723570-92c248c8-d74d-44a7-a761-10031c336cd2.webm)

After: 

[after-dropdown.webm](https://user-images.githubusercontent.com/12435965/209723645-194cd91a-3aca-4c95-a58f-0aeb7a43dea1.webm)